apps/api/src/app/shared/interceptors/product-feature.interceptor.ts

```diff
@@ -0,0 +1,51 @@
+import { CallHandler, ExecutionContext, HttpException, Injectable, NestInterceptor } from '@nestjs/common';
+import { Reflector } from '@nestjs/core';
+import { OrganizationRepository } from '@novu/dal';
+import {
+  ApiServiceLevelEnum,
+  IJwtPayload,
+  productFeatureEnabledForServiceLevel,
+  ProductFeatureKeyEnum,
+} from '@novu/shared';
+import { Observable } from 'rxjs';
+import { ProductFeature } from '../decorators/product-feature.decorator';
+
+@Injectable()
+export class ProductFeatureInterceptor implements NestInterceptor {
+  constructor(private reflector: Reflector, private organizationRepository: OrganizationRepository) {}
+
+  async intercept(context: ExecutionContext, next: CallHandler): Promise<Observable<any>> {
+    try {
+      const user = this.getReqUser(context);
+
+      const { organizationId } = user;
+
+      const organization = await this.organizationRepository.findById(organizationId);
```

Comment:
We should probably throw an error when organization is not found, and remove the optional chaining below.